browser/registry.go

```diff
@@ -279,7 +278,12 @@ func (r *browserRegistry) handleIterEvents(eventsCh <-chan *k6event.Event, unsub
 			// we have to initialize traces registry on the first VU iteration
 			// so we can get access to the k6 TracerProvider.
 			r.initTracesRegistry()
-			tracedCtx := r.tr.startIterationTrace(ctx, data)
+
+			// Wrap the tracer into the browser context to make it accessible for the other
+			// components that inherit the context so these can use it to trace their actions.
+			tracerCtx := common.WithTracer(context.Background(), r.tr.tracer)
```

Comment:
I'm wondering if this should use `vuCtx`?

Reply:
We can't use `vuCtx` as its lifecycle is shorter than the iteration. Instead, we use background context for the browser, and then stop it based on the `iterEnd` event.

Reply:
That makes sense, see [a355a21](https://github.com/grafana/xk6-browser/pull/1100/commits/a355a21b2cf0e2c9e6c681bb97f394e9ff61dd31).

Reply:
🙇